### PR TITLE
Update gitlab token docs

### LINKED ISSUE #169: Question: do we really need a Gitlab personal access token for Gitlab ci?
I am currious to understand why this is a requirement for Gitlab because I used the built in [gitlab ci release](https://docs.gitlab.com/ee/user/project/releases/release_cicd_examples.html#create-release-metadata-in-a-custom-script) tooling and it doesn't need a token to create a release.

Based on their code they use the `CI_JOB_TOKEN` instead: https://gitlab.com/gitlab-org/release-cli

I am not familiar with the inner working of the project yet so I was wondering if this was still a requirement. The `CI_JOB_TOKEN` is a recent addition so I can imagine that maybe it could works as well.

### PR DESCRIPTION
**Pull request by cliedeman**
Resolves #169

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,13 +96,20 @@ jobs:
...
 ### Example GitLab CI Config
 
-#### GitLab token
-It is necessary to create a new Gitlab personal access token with the `api` scope [here](https://gitlab.com/profile/personal_access_tokens).
+#### Job Token
+If you do not provide a PAT the [job token](https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html) will be used.
+This restricted token can create releases but not read commits. The [git strategy](https://docs.gitlab.com/ee/ci/runners/configure_runners.html#git-strategy) must be set to clone so that we can read the commits from the repository. See example below
+
+#### Personal Access Token
+Create a new Gitlab personal access token with the `api` scope [here](https://gitlab.com/profile/personal_access_tokens).
```
Comment (jamietanna) [inline README.md:104]: That link doesn't seem to work for me, but this does:

```suggestion
Create a new Gitlab personal access token with the `api` scope [here](https://gitlab.com/-/user_settings/personal_access_tokens).
```

Comment (jamietanna) [inline README.md:104]: Additionally, we can prefill the scope:

```suggestion
Create a new Gitlab personal access token with the `api` scope [here](https://gitlab.com/profile/personal_access_tokens?name=go-semantic-release+for+example%2Frepo&scopes=api).
```

Comment (christophwitzko) [reply README.md]: Thanks for this @jamietanna

**Review by christophwitzko — APPROVED** [review]
Thanks ✨

**Pull request closed without merging**